src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,271 @@
+            Task currentTask = tasks.get(taskIndex);
+            System.out.println(currentTask.getTaskType() + currentTask.getStatusIcon() + " "
+                    + currentTask.getDescription());
+            tasks.remove(taskIndex);
+            if (tasks.size() > 1) {
+                System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+            } else  {
+                System.out.println("Now you have " + tasks.size() + " task in the list.");
+            }
+        }
+    }
+
+    public static void loadData() {
+        try {
+            Path dataFilePath = Paths.get("data/");
+            Files.createDirectories(dataFilePath);
+            File data = new File ("data/duke.txt");
+            Scanner readFile = new Scanner (data);
+            while (readFile.hasNextLine()) {
+                String input = readFile.nextLine();
+                //split string by whitespace + "|" + whitespace
+                String[] taskSplit = input.split("\\s\\|\\s");
+                if (taskSplit[0].equals("[D]")) {
+                    Deadline d = new Deadline (taskSplit[2], taskSplit[3]);
+                    tasks.add(d);
+                    if (taskSplit[1].equals("1")) {
+                        d.setDone();
+                    }
+                } else if (taskSplit[0].equals("[T]")) {
+                    Todo t = new Todo (taskSplit[2]);
+                    tasks.add(t);
+                    if (taskSplit[1].equals("1")) {
+                        t.setDone();
+                    }
+                } else if (taskSplit[0].equals("[E]")) {
+                    Event e = new Event (taskSplit[2], taskSplit[3]);
+                    tasks.add(e);
+                    if (taskSplit[1].equals("1")) {
+                        e.setDone();
+                    }
+                }
+            }
+            readFile.close();
+        } catch (IOException e) {
+            System.out.println("Error when creating data directory. Please try again.");
+        }
+    }
+
+    public static boolean saveData() {
+        try {
+
+            System.out.println("Saving your data...");
+            FileWriter fileWriter = new FileWriter("data/duke.txt");
+            for (int i = 1; i < tasks.size(); i++) {
+                Task currentTask = tasks.get(i);
+                fileWriter.write(currentTask.getTaskType() + " | " + ((currentTask.getIsDone()) ? "1" : "0") + " | "
+                        + currentTask.getDescription() + ((currentTask.hasDateTime()) ? " | "
+                        + currentTask.getDateTime() : "") + "\n" );
+            }
+            fileWriter.close();
+            System.out.println("Data saved");
+            return true;
+        } catch (IOException e) {
+            System.out.println("Error occurred when saving your data. Please try again.");
+            return false;
+        }
+    }
+
+    public static void getInput() {
+        while (true) {
```

Comment:
There is very high level of indentation in this method, try structuring your logic around the happy path to make the code more readable and understandable.